File: Application/DTOs/Checkout/GuestCheckoutRequest.cs
using Common.Constants;
using System.ComponentModel.DataAnnotations;

namespace Application.DTOs.Checkout;

public class GuestCheckoutRequest
{
    [Required]
    [StringLength(100, MinimumLength = 1)]
    public string Name { get; set; } = null!;

    [Required]
    [EmailAddress]
    [StringLength(256)]
    public string Email { get; set; } = null!;

    [Required]
    [StringLength(32, MinimumLength = 1)]
    public string PhoneNumber { get; set; } = null!;

    [Required]
    [MinLength(1)]
    public string DeliveryAddress { get; set; } = null!;

    [Range(-90, 90)]
    public double? DeliveryLatitude { get; set; }

    [Range(-180, 180)]
    public double? DeliveryLongitude { get; set; }

    [Required]
    public bool PayOnline { get; set; } = false;

    [Required]
    [MinLength(1)]
    [MaxLength(OrderValidationConstants.MaxOrderLineCount)]
    public List<GuestCheckoutItemRequest> Items { get; set; } = new();
}

File: WebApp/src/app/features/checkout/models/guest-checkout-request.ts
export const MAX_GUEST_CHECKOUT_LINE_COUNT = 50;
export const MAX_GUEST_CHECKOUT_TOTAL_QUANTITY = 200;

export interface GuestCheckoutItemRequest {
  readonly menuItemId: string;
  readonly quantity: number;
}

export interface GuestCheckoutRequest {
  readonly name: string;
  readonly email: string;
  readonly phoneNumber: string;
  readonly deliveryAddress: string;
  readonly deliveryLatitude: number | null;
  readonly deliveryLongitude: number | null;
  readonly payOnline: boolean;
  readonly items: GuestCheckoutItemRequest[];
}
